### ISSUE TITLE
ARM template variable copy block highlighting error

### ISSUE THREAD

**Issue by elentar** [microsoft/vscode-azurearmtools#30]
There is a syntax highlighting error in the azurerm-vscode-tools extension when editing ARM templates. In the following block in the variables section, the opening "copy": block is underlined green indicating that it is never used and all references to the name "generatedSecurityRules" are underlined red indicating that it is not defined. The template actually executes fine when run against the Azure API.

```
"copy": [
            {
                "name": "generatedSecurityRules",
                "count": "[length(parameters('appServiceRules'))]",
                "input": {
                    "name": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].name]",
                    "properties": {
                        "description": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].description]",
                        "protocol": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].protocol]",
                        "sourcePortRange": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].sourcePortRange]",
                        "destinationPortRange": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].destinationPortRange]",
                        "destinationPortRanges": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].destinationPortRanges]",
                        "sourceAddressPrefixes": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].sourceAddressPrefixes]",
                        "destinationAddressPrefixes": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].destinationAddressPrefixes]",
                        "access": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].access]",
                        "priority": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].priority]",
                        "direction": "[parameters('appServiceRules')[copyindex('generatedSecurityRules')].direction]"
                    }
                }
            }
        ],
        "securityRuleList": "[concat(variables('generatedSecurityRules'), parameters('denyInternetRule'))]"
```

**Comment by StephenWeatherford** [comment]
See https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-templates-variables for how this is supposed to work.

**Comment by VHamar** [comment]
Example file that shows the highlight errors below. 

"testobj": "[variables('obj').**_array_**[copyIndex()].bar]",
"disk": "[variables('**_disks_**')[copyIndex('disklist')].disk]"

```json
{
    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
    "contentVersion": "1.0.0.0",
    "parameters": {},
    "variables": {
        "normal": "foo",
        "count": 3,
        "obj": {
            "copy": [
                {
                    "name": "array",
                    "count": "variables('count')",
                    "input": {
                        "bar": "[concat('foobar',copyIndex('array'))]"
                    }
                }
            ]
        },
        "copy": [
            {
                "name": "disks",
                "count": "variables('count')",
                "input": {
                    "disk": "[concat('lun',copyIndex('array'))]"
                }
            }
        ]
    },
    "resources": [
        {
            "normalVar": "[variables('normal')]"
        },
        {
            "testobj": "[variables('obj').array[copyIndex()].bar]",
            "copy": {
                "name": "objarraycopy",
                "count": "[variables('count')]"
            }
        },
        {
            "diskinfo": {
                "copy": [
                    {
                        "name": "disklist",
                        "count": "[variables('count')]",
                        "input": {
                            "diskid": "[copyIndex('disklist')]",
                            "disk": "[variables('disks')[copyIndex('disklist')].disk]"
                        }
                    }
                ]
            }
        }
    ],
    "outputs": {
        
    }
}
```

**Comment by andyKalman** [comment]
I can 2nd this one as well. In VS this is not the behavior FWIW........

**Comment by fgauna12** [comment]
I third this 😄

**Comment by StephenWeatherford** [comment]
Good to know, thanks. It may take me a while to get to it, though.

**Comment by samhodgkinson** [comment]
Just come across this issues. Although its not a massive problem, it does make it hard to review and edit code. Any news on when this might be fixed.

**Comment by StephenWeatherford** [comment]
I'm hoping to get to it in a month or two, but we've got a lot going on, so I can't promise it.

**Comment by hanuravim** [comment]
Any news on this please, resolved/on-going ?

**Comment by StephenWeatherford** [comment]
Still on the list, hoping a month or two.

**Comment by StephenWeatherford** [comment]
Coming soon.

**Comment by StephenWeatherford** [comment]
Fixed by #367

**Comment by neilpeterson** [comment]
Fix went out today with this fix, thanks.
